### DISCUSSION TITLE
Conditional `mul_add` chunk gen speed up.

### DISCUSSION THREAD

**Discussion by kralverde** [Pumpkin-MC/Pumpkin#400] [Ideas]
@julianbraha I saw you made the original PR for using `mul_add` to speed up float operations in chunk gen. Those we ultimately reverted because that function was extremely slow on non-FMA cpus.

I was wondering if you had access to a FMA cpu and could run benchmarks to see if it would be worth adding a compile time method to use `mul_add` or `x * a + b` based on whether the CPU supports.

**Comment by vyPal** [comment]
Maybe something like this could be used: https://crates.io/crates/multiversion

**Reply by kralverde** [reply]
A simple cfg check should suffice for this
